### PR TITLE
Changes for38

### PR DESCRIPTION
**Pull request by prateekbh**
- fixed switch
- implemented tab bar

### CODE REVIEW

**Comment by cromefire** [comment]
@prateekbh please run `npm run tslint:fix` and `npm run tslint` before commits. This also runs/fixes prettier

**Comment by cromefire** [comment]
No need to do this manually

**Comment by cromefire** [comment]
Travis enforces tslint style BTW

File: ts/TabBar/index.tsx [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+  protected materialDom(props) {
+    return (
+      <span className="material-icons" {...props}>
+        {props.children}
+      </span>
+    );
+  }
+}
+
+export interface ITabProps extends JSX.HTMLAttributes {
```
Comment (cromefire) [inline ts/TabBar/index.tsx]: No need for `extends JSX.HTMLAttributes` for components that extend `MaterialComponent` they do that by them selves.

Comment (prateekbh) [reply ts/TabBar/index.tsx]: done

**Comment by cromefire** [comment]
Seems good

File: docs/src/routes/tabbar/index.js [hunk 2]
```diff
@@ -0,0 +1,130 @@
+// Dependencies
+import {h, Component} from 'preact';
+
+// Material Components
+import TabBar from '../../../../TabBar';
+import Icon from '../../../../Icon';
+import style from './style';
```
[the remark this answers is not in the document]
Comment (prateekbh) [reply docs/src/routes/tabbar/index.js]: yep

**Comment by artipost** [comment]
Failed pictures (1): [failed-pictures.zip](https://gist.githubusercontent.com/artipost/4e4dae8e1ee26e22518e59e2977276dd/raw/93717e3065427e9a84117bdab2f418c1e10adc23/failed-pictures.zip)

##
:postbox: Posted by [artipost.io](https://artipost.io), through Travis build [414220186](https://travis-ci.org/prateekbh/preact-material-components/builds/414220186), for commit [478a7a8](https://github.com/prateekbh/preact-material-components/pull/923/commits/478a7a8caf1725bfeb42088936d2b8d6354a99f3).

**Comment by prateekbh** [comment]
@cromefire I am willing to do a full-fledged 1.5.1 release WDYT?

**Comment by cromefire** [comment]
I think we should go `beta` for like 1 or 2 days, for people to try it and that I have time to improve the docs.

File: docs/src/routes/tabbar/index.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (cromefire) [reply docs/src/routes/tabbar/index.js]: okay just mentioning

**Comment by prateekbh** [comment]
done

**Comment by cromefire** [comment]
Okay I will look at the docs tomorrow (CEST).

**Pull request merged**